### PR TITLE
Infer new request method

### PR DESCRIPTION
**Pull request by bszmelcz**
### Details:
 - [x] add infer_new_request function
 - [x] add tests for infer_new_request
 - [x] remove redundant leftover from Blob

### Tickets:
 - 68621

### CODE REVIEW

File: runtime/bindings/python/src/pyopenvino/core/executable_network.cpp [hunk 1]
```diff
@@ -21,26 +24,52 @@ void regclass_ExecutableNetwork(py::module m) {
...
+            Containers::InferResults results;
+            for (const auto out : self.outputs()) {
+                results.push_back(request.get_tensor(out));
+            }
+            return results;
         },
-        py::arg("name"));
+        py::arg("inputs"));
 
-    cls.def(
```
Comment (akuporos) [inline runtime/bindings/python/src/pyopenvino/core/executable_network.cpp:38]: why have you commented this method?

File: runtime/bindings/python/tests/test_inference_engine/test_executable_network.py [hunk 2]
```diff
@@ -236,3 +218,60 @@ def test_inputs_docs(device):
...
+def test_infer_tensor_wrong_input_data(device):
+    ie = Core()
+    func = ie.read_model(model=test_net_xml, weights=test_net_bin)
+    img = read_image()
+    img = np.ascontiguousarray(img)
+    tensor = Tensor(img, shared_memory=True)
+    exec_net = ie.compile_model(func, device)
+    with pytest.raises(TypeError) as e:
+        exec_net.infer_new_request({4.5: tensor})
+    assert "Incompatible key type!" in str(e.value)
```
Comment (akuporos) [inline runtime/bindings/python/tests/test_inference_engine/test_executable_network.py:277]: please, add blank line after this line

File: runtime/bindings/python/tests/test_inference_engine/test_executable_network.py [hunk 3]
```diff
@@ -27,25 +28,6 @@ def read_image():
     return image
 
 
-def test_get_metric(device):
```
Comment (akuporos) [inline runtime/bindings/python/tests/test_inference_engine/test_executable_network.py:30]: Why have you removed this test?

**Comment by akuporos** [comment]
One more question which probably may help us -- have you tried to run tests on your machine from ```openvino/runtime/bindgins/tests``` directory? Did they pass?

File: runtime/bindings/python/tests/test_inference_engine/test_executable_network.py (hunk 3 above)
Comment (bszmelcz) [reply runtime/bindings/python/tests/test_inference_engine/test_executable_network.py:30]: this same as with methods

File: runtime/bindings/python/src/pyopenvino/core/executable_network.cpp (hunk 1 above)
Comment (bszmelcz) [reply runtime/bindings/python/src/pyopenvino/core/executable_network.cpp:38]: I was testing something and forgot to revert the changes, I will undo them in a second


> One more question which probably may help us -- have you tried to run tests on your machine from `openvino/runtime/bindgins/tests` directory? Did they pass?

I was locally runnig tests only for tests/test_inference_engine

**Comment by bszmelcz** [comment]
> One more question which probably may help us -- have you tried to run tests on your machine from `openvino/runtime/bindgins/tests` directory? Did they pass?

Locally I was runnig tests only from tests/test_inference_engine dir

File: runtime/bindings/python/src/openvino/ie_api.py [hunk 4]
```diff
@@ -45,6 +46,11 @@ def infer(request: InferRequest, inputs: dict = None) -> np.ndarray:
     res = request._infer(inputs=normalize_inputs(inputs if inputs is not None else {}))
     return np.asarray([copy.deepcopy(tensor.data) for tensor in res])
 
+
+def infer_new_request(exec_net: ExecutableNetwork, inputs: dict = None) -> np.ndarray:
+    res = exec_net._infer_new_request(inputs=normalize_inputs(inputs if inputs is not None else {}))
+    return np.asarray([copy.deepcopy(tensor.data) for tensor in res])
```
Comment (jiwaszki) [inline runtime/bindings/python/src/openvino/ie_api.py:52]: Remove `np.asarray`, I recently found out that this approach broke some of the cases where outputs can differ in sizes.

File: runtime/bindings/python/src/openvino/ie_api.py [hunk 5]
```diff
@@ -104,11 +110,6 @@ def __new__(cls, tensor_desc: TensorDesc, arr: np.ndarray = None):  # type: igno
         else:
             raise AttributeError(f"Unsupported precision {precision} for Blob")
 
-# flake8: noqa: D102
-def blob_from_file(path_to_bin_file: str) -> BlobWrapper:
```
Comment (jiwaszki) [inline runtime/bindings/python/src/openvino/ie_api.py:108]: If you remove all blob related funcs, please remove `BlobWrapper` and `Blob` pybind related files as well.

**Review by jiwaszki — CHANGES_REQUESTED** [review]

File: runtime/bindings/python/src/openvino/ie_api.py (hunk 5 above)
Comment (bszmelcz) [reply runtime/bindings/python/src/openvino/ie_api.py:108]: I wanted to remove all of them but ie_process_info.cpp in core still uses Blob so I removed all the functions that are not used anywhere

File: runtime/bindings/python/src/openvino/ie_api.py (hunk 4 above)
Comment (bszmelcz) [reply runtime/bindings/python/src/openvino/ie_api.py:52]: done

File: runtime/bindings/python/src/openvino/ie_api.py (hunk 5 above)
Comment (akuporos) [reply runtime/bindings/python/src/openvino/ie_api.py:108]: you may remove ```ie_preprocess_info```, it's useless now because it comes from old API

**Comment by akuporos** [comment]
BTW,  @jiwaszki  may be we will remove Blob and so one in a separate PR? This one includes ```infer_new_request``` method which is really important while removing all Blob using may cause a lot of changes and red CI

**Comment by bszmelcz** [comment]
> BTW, @jiwaszki may be we will remove Blob and so one in a separate PR? This one includes `infer_new_request` method which is really important while removing all Blob using may cause a lot of changes and red CI

Do you want me then to revert the changes with removing several of blob related functions?

**Comment by akuporos** [comment]
> > BTW, @jiwaszki may be we will remove Blob and so one in a separate PR? This one includes `infer_new_request` method which is really important while removing all Blob using may cause a lot of changes and red CI
> 
> Do you want me then to revert the changes with removing several of blob related functions?

I think it's better to revert the changes with removal. Please, also update your branch with develop branch.

File: runtime/bindings/python/src/openvino/ie_api.py (hunk 5 above)
Comment (bszmelcz) [reply runtime/bindings/python/src/openvino/ie_api.py]: revert the removal of the Blob components in this PR

**Review by akuporos — APPROVED** [review]

**Review by jiwaszki — APPROVED** [review]

**Pull request merged**